src/main/java/org/aksw/simba/squirrel/components/WorkerComponent.java

```diff
@@ -154,9 +146,15 @@ public void addNewUris(List<CrawleableUri> uris) {
     @Override
     public void crawlingDone(List<CrawleableUri> crawledUris, List<CrawleableUri> newUris) {
         try {
-            sender.sendData(serializer.serialize(new CrawlingResult(crawledUris, newUris, worker.getId())));
+            senderFrontier.sendData(serializer.serialize(new CrawlingResult(crawledUris, newUris, worker.getId())));
+
+            if (DeduplicatorComponent.DEDUPLICATION_ACTIVE) {
```

Comment:
@MichaelRoeder ok, but ``DeduplicatorComponent``is subclass of ``AbstractComponent``, therefore it is never instantiated in our code and we have no reference to the object. If we then have a non-static member, how could be access it from outside?

Reply:
You could use an environmental variable with which this can be configured "from outside", i.e., before starting the component:

```java
    public static final String DEDUPLICATION_ACTIVE_KEY = "DEDUPLICATION_ACTIVE";
    public static final boolean DEDUPLICATION_ACTIVE_DEFAULT = false;

    protected boolean deduplicationActive = DEDUPLICATION_ACTIVE_DEFAULT;

    public void init() throws Exception {
        super.init();
        Map<String, String> env = System.getenv();
        if (env.containsKey(DEDUPLICATION_ACTIVE_KEY)) {
            try {
                deduplicationActive = Boolean.parseValue(env.get(DEDUPLICATION_ACTIVE_KEY));
            } catch (Exception e){
                // handle error
            }
        }
    }
```